router/pkg/mcpserver/strings.go

```diff
@@ -0,0 +1,47 @@
+package mcpserver
+
+import (
+	"strings"
+	"unicode"
+)
+
+// toSnakeCase converts a string to snake_case format
+// It handles various input formats:
+// - CamelCase -> camel_case
+// - PascalCase -> pascal_case
+// - Space-separated -> space_separated
+// - Hyphen-separated -> hyphen_separated
+// Acronyms like "URL" or "API" are preserved as lowercase without internal underscores
+func toSnakeCase(s string) string {
```

Comment:
is there no libary for this?

Reply:
There is but I was a bit reluctant to pull a library for this in the first place. I can do it no objection anymore.